New tutorial: Add `zeekctl` section and cleanup the remainder (#5084)
zeek/zeek: doc/tutorial/invoking-zeekctl.rst

.. _ZeekControl documentation: https://github.com/zeek/zeekctl
.. _ZeekControl cron: https://github.com/zeek/zeekctl?tab=readme-ov-file#zeekcontrol-cron-command

.. _invoking-zeekctl:

==============
 ZeekControl
==============

ZeekControl (or ``zeekctl``) is Zeek's primary tool for cluster orchestration.
In this case, a Zeek cluster is simply multiple Zeek processes which analyze

> network traffic. Nodes within a cluster have different jobs. You use

Reviewer: ```suggestion
network traffic. Nodes within a cluster have different roles. You use
```

   worker-2     worker  localhost        running   4512   08 Dec 21:47:35
   # zeekctl start proxy-1
   starting proxy ...

   # zeekctl status
   Name         Type    Host             Status    Pid    Started
   logger-1     logger  localhost        running   4341   08 Dec 21:47:32
   manager      manager localhost        running   4394   08 Dec 21:47:33
   proxy-1      proxy   localhost        running   4947   08 Dec 21:53:49
   worker-1     worker  localhost        running   4511   08 Dec 21:47:35
   worker-2     worker  localhost        running   4512   08 Dec 21:47:35

This requires manually intervention. Instead, you can use ``zeekctl`` with
``cron`` in order to automatically check for crashed nodes and restart them.
See the `ZeekControl cron`_ command reference for more information.

Many basic clusters can be maintained with just what was discussed here:
a mixture of ``zeekctl`` commands like ``deploy``, ``start``, ``stop``,
and ``status``. You can use ``top`` and ``cron`` for extra visibility
or configuration on your cluster. Then, configure the cluster according
to your particular environment, and that should be what you need to get

> a capable Zeek cluster.

Reviewer: For examining if anything is off,  `zeekctl diag [node]` is the one command to use. Might be worth including. It shows `.stderr` and also `reporter.log` by default IIRC.

.. _ZeekControl documentation: https://github.com/zeek/zeekctl
.. _ZeekControl cron: https://github.com/zeek/zeekctl?tab=readme-ov-file#zeekcontrol-cron-command

.. _invoking-zeekctl:

==============
 ZeekControl
==============

> ZeekControl (or ``zeekctl``) is Zeek's primary tool for cluster orchestration.

Reviewer: I have used "Zeekctl" in the NEWS recently, but I like the idea of using ZeekControl as "the spelling" when it's not about invoking the command.

   # kill 4446
   # zeekctl status
   Name         Type    Host             Status    Pid    Started
   logger-1     logger  localhost        running   4341   08 Dec 21:47:32
   manager      manager localhost        running   4394   08 Dec 21:47:33
   proxy-1      proxy   localhost        crashed
   worker-1     worker  localhost        running   4511   08 Dec 21:47:35
   worker-2     worker  localhost        running   4512   08 Dec 21:47:35
   # zeekctl start proxy-1
   starting proxy ...

   # zeekctl status
   Name         Type    Host             Status    Pid    Started
   logger-1     logger  localhost        running   4341   08 Dec 21:47:32
   manager      manager localhost        running   4394   08 Dec 21:47:33
   proxy-1      proxy   localhost        running   4947   08 Dec 21:53:49
   worker-1     worker  localhost        running   4511   08 Dec 21:47:35
   worker-2     worker  localhost        running   4512   08 Dec 21:47:35

This requires manually intervention. Instead, you can use ``zeekctl`` with
``cron`` in order to automatically check for crashed nodes and restart them.
See the `ZeekControl cron`_ command reference for more information.

.. note::

>    ZeekControl itself has no active monitoring. Therefore, in order to

Reviewer: ```suggestion
   ZeekControl itself has no active process monitoring. Therefore, in order to
```

   perform automatic restarts, you need to put the ``zeekctl cron``

>    command into a ``crontab`` entry.

Reviewer: I there a nice one-liner? Like totally untested:
```
cat << EOF > /etc/cron.d/zeek-cron
# Run zeekctl cron every minute
* * * * * root /usr/local/zeek/bin/zeekctl cron
EOF
```

If this is all in a container, there's probably no `crond` running, so not sure how useful that is without explaining even more...

Many basic clusters can be maintained with just what was discussed here:
a mixture of ``zeekctl`` commands like ``deploy``, ``start``, ``stop``,
and ``status``. You can use ``top`` and ``cron`` for extra visibility
or configuration on your cluster. Then, configure the cluster according
to your particular environment, and that should be what you need to get

> a capable Zeek cluster.

Reviewer: Ah, that comes in `logs.rst`, might still be good to do a forward reference?

.. _invoking-zeekctl:

.. _zeekcontrol cron: https://github.com/zeek/zeekctl?tab=readme-ov-file#zeekcontrol-cron-command

.. _zeekcontrol documentation: https://github.com/zeek/zeekctl

#############
 ZeekControl
#############

ZeekControl (or ``zeekctl``) is Zeek's primary tool for cluster

> orchestration. In this case, a Zeek cluster is simply multiple Zeek

Reviewer: I wonder if we could go fancy/abstract, but ignore if that seems to be too abstract.
```suggestion
orchestration. Technically, a Zeek cluster is a set of independent Zeek processes that analyze network traffic.
```

processes which analyze network traffic. Every Zeek process in a cluster

> is called a "node"---nodes within a cluster have different roles. A

Reviewer: ```suggestion
is called a "node". Nodes within a cluster have different roles. A
```

Reviewer: Makes sense.
